static/js/components/EducationDialog.js

```diff
@@ -23,11 +26,16 @@ export default class EducationDialog extends ProfileFormFields {
   }
   educationLevelLabels: Object;
 
-  static propTypes = {
-    open:           React.PropTypes.bool,
-    onClose:        React.PropTypes.func,
-    onSave:         React.PropTypes.func,
-    showLevelForm:  React.PropTypes.bool,
+  props: {
+    ui:                           UIState,
+    setEducationDialogVisibility: () => void,
+    setEducationDegreeLevel:      () => void,
+    setEducationDialogIndex:      () => void,
+    clearProfileEdit:             () => void,
+    saveProfile:                  BoundSaveProfile,
```

Comment:
re: 'BoundSaveProfile', sounds good 👍 

re: `SetWorkHistoryEdit` and `SetEducationDegreeInclusions`, i see how it's valuable to insist that those props will be functions that return a Promise. that way we can invoke `setEducationDegreeInclusions` as a function, and we know that we can use `.then()` after the invocation. can we declare a type called `Dispatcher` or something that's vague enough to say that it's a function that returns a Promise and not worry about the exact arguments? i really don't agree with declaring a custom named type for each of these functions that are used in exactly one place. it bloats the codebase with relatively little added value, and like i suggested above it sets a precedent for creating a custom type for every single function we create. what if we change other functions like `setEducationDialogIndex` or `setEducationDegreeLevel` to return the Promise from `dispatch`? will we declare two more types then? if you have more questions about this, let's chat in person.